plugin/pkg/admission/disableservicelinks/admission.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disableservicelinks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+// PluginName indicates name of admission plugin.
+const PluginName = "DisableServiceLinks"
+
+// Register is called by the apiserver to register the plugin factory.
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return newDisableServiceLinks(), nil
+	})
+}
+
+// newDisableServiceLinks creates a new instance of the DisableServiceLinks admission controller.
+func newDisableServiceLinks() *plugin {
+	return &plugin{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
```

Comment:
I think this should operate only on Create, IIUIC Updates to a Pod will not recreate the environment variables

Reply:
Good Catch! 

I did some testing in kind and found those fields are immutable when updating a pod. I removed `admission.Update` in my commit.


```yaml
$ kubectl apply -f simple-pod.yaml 
The Pod "nginx" is invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`,`spec.initContainers[*].image`,`spec.activeDeadlineSeconds`,`spec.tolerations` (only additions to existing tolerations),`spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)
  core.PodSpec{
  	... // 26 identical fields
  	RuntimeClassName:          nil,
  	Overhead:                  nil,
- 	EnableServiceLinks:        &true,
+ 	EnableServiceLinks:        &false,
  	TopologySpreadConstraints: nil,
  	OS:                        nil,
  	... // 2 identical fields
  }



```

Reply:
Supporting update might be a no-op today, but could be future-safe